From 85a34517f2f67e67c10728a9a9a37c14fba51092 Mon Sep 17 00:00:00 2001
From: Ran Wang <ran.wang_1@nxp.com>
Date: Fri, 3 Nov 2017 10:42:04 +0800
Subject: [PATCH 186/211] usb: Add delay to fix USB 2.0 stick enumeration failure on LS208xA

Force delay 10ms between each control messages to fix Transcend
and Kingston DT 101 G2 USB2.0 stick fail to be enumerated by LS208xA.

Signed-off-by: Ran Wang <ran.wang_1@nxp.com>
---
 common/usb.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/common/usb.c b/common/usb.c
index 0904259..eeacc80 100644
--- a/common/usb.c
+++ b/common/usb.c
@@ -223,6 +223,9 @@ int usb_control_msg(struct usb_device *dev, unsigned int pipe,
 		return -EINVAL;
 	}
 
+	/* Add delay to support more devices' enumeration */
+	mdelay(10);
+
 	/* set setup command */
 	setup_packet->requesttype = requesttype;
 	setup_packet->request = request;
-- 
1.7.1

